src/odemis/acq/fastem.py

```diff
@@ -783,31 +785,33 @@ def correct_beam_shift(self):
         good beam positions we calculate in what direction and how much to shift beams, such that they are always
         centered on the mppc detector.
         """
+        pixel_size = self._ccd.pixelSize.value
+        magnification = self._lens.magnification.value
+        refractive_index = self._lens.refractiveIndex.value
+        numerical_aperture = self._lens.numericalAperture.value
+        wavelength = 550e-9
+        sigma = (magnification / pixel_size) * synthetic.psf_sigma_wffm(
+            refractive_index, numerical_aperture, wavelength
+        )
         # asap=False: wait until new image is acquired (don't read from buffer)
         ccd_image = self._ccd.data.get(asap=False)
-
-        # Find the location of the spots on the diagnostic camera.
-        spot_coordinates, *_ = FindGridSpots(ccd_image, (8, 8))
-
-        # Transform the spots from the diagnostic camera coordinate system to a right-handed coordinate
-        # system with the origin in the bottom left.
-        spot_coordinates[:, 1] = ccd_image.shape[1] - spot_coordinates[:, 1]  # [px]
+        tform = estimate_grid_orientation_from_img(ccd_image, (8, 8), AffineTransform, sigma)
```

Comment:
@effting we were using an `AffineTransform` in `FindGridSpots`, I see that in your code you use a `SimilarityTransform`. Did you try both for the tests with missing spots, and was one performing "better"?

Reply:
The main goal of the beamshift correction is to control (limit) cross-talk. We need to project the image of the multi-probe pattern on the MPPC sensor array. We can only control the scale, rotation, and translation so it makes sense to me to not fit the anisotropic scaling and shear here.